### PR TITLE
scan qr address with prefix

### PR DESCRIPTION
**Pull request by MouazAlzahabi**
Handles #1584

### CODE REVIEW

File: Multisig/Data/Services/Utils/GSError.swift [hunk 1]
```diff
@@ -242,6 +242,15 @@ enum GSError {
         let loggable = false
     }
 
+    struct AddressMismatchNetwork: DetailedLocalizedError {
+        let description = "Can’t use this address"
+        let reason = "This address is not matching the selected network."
+        let howToFix = "Please use different address or switch network"
```
Comment (sche) [inline Multisig/Data/Services/Utils/GSError.swift]: "Please use **a** different address or switch network" - checked with Grammarly :)

File: Shared/Utils/Address.swift [hunk 2]
```diff
@@ -95,13 +95,11 @@ struct Address: Hashable, ExpressibleByStringInterpolation, CustomStringConverti
...
 
-    init(_ value: String, isERC681: Bool) throws {
-        var addressString = value
-        if isERC681 {
-            addressString = Self.addressFromERC681(addressString)
-        }
-
-        _store = try EthereumAddress(hex: addressString, eip55: false)
+    // This will check if ERC681 or EIP3770
+    static func addressWithPrefix(text: String) throws -> (prefix: String?, address: Address) {
```
Comment (sche) [inline Shared/Utils/Address.swift]: Would it be better to make prefix as a property of `Address`?

Comment (sche) [reply Shared/Utils/Address.swift]: I am also not sure how do we add prefixes to addresses where we need to present them in the UI. I would assume it to be a part of the Address description if it has a prefix set.

Comment (MouazAlzahabi) [reply Shared/Utils/Address.swift]: as prefix is the short name of corresponding network, is taken from there now

**Pull request merged**